cmd/mole/main.go

```diff
@@ -57,7 +74,90 @@ func main() {
 	}
 }
 
+func stopDaemon(app cli.App) error {
+	daemonDir := fmt.Sprintf("%s/%s", instancesDir, app.Stop)
+	pidPathName := fmt.Sprintf("%s/pid", daemonDir)
+	logPathName := fmt.Sprintf("%s/mole.log", daemonDir)
+
+	cntxt := &daemon.Context{
+		PidFileName: pidPathName,
+		PidFilePerm: 0644,
+		LogFileName: logPathName,
+		LogFilePerm: 0640,
+		Umask:       027,
+		Args:        os.Args,
+	}
+
+	d, err := cntxt.Search()
+
+	err = d.Kill()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createInstancesDir() error {
+	_, err := os.Stat(instancesDir)
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(instancesDir, 0755)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func startDaemonProcess(aliasName string) error {
+	cntxt := &daemon.Context{
+		PidFileName: "pid",
+		PidFilePerm: 0644,
+		LogFileName: "mole.log",
+		LogFilePerm: 0640,
+		Umask:       027,
+		Args:        os.Args,
+	}
+	d, err := cntxt.Reborn()
+	if err != nil {
+		return err
+	}
+	if d != nil {
+		daemonDir := fmt.Sprintf("%s/%s", instancesDir, aliasName)
```

Comment:
Why the instance filesystem can't be created before `cntxt.Reborn()` to be called? Would that avoid the `os.Rename` calls?

Reply:
Because daemon process will execute this code again, so i think it should be responsibility of  root process to do create directory and move file to .mole > instances > alias_name